### PR TITLE
Fix `wchar` unit test on FreeBSD

### PR DESCRIPTION
**Pull request by krader1961**
Part of resolving #577

### CODE REVIEW

File: src/cmd/ksh93/tests/wchar.sh [hunk 1]
```diff
@@ -21,7 +21,13 @@
 # Tests for \u[...] and \w[...] input and output
 #
 
-locales="en_US.UTF-8 en_US.utf8 en_US.ISO-8859-15 en_US.iso885915 zh_CN.GB18030 zh_CN.gb18030"
+locales="en_US.UTF-8 en_US.utf8 en_US.ISO-8859-15 en_US.iso885915"
+# On FreeBSD 11.1 the native locale subsystem cannot convert the Euro currency symbol from UTF-8 to
+# GB1803. See https://github.com/att/ast/issues/577#issuecomment-396113061.
+if [[ $OS_NAME != "FreeBSD" ]]
+then
+    locales="$locales zh_CN.GB18030 zh_CN.gb18030"
```
Comment (siteshwar) [inline src/cmd/ksh93/tests/wchar.sh]: Are the locales `zh_CN.GB18030` and `zh_CN.gb18030` supported on FreeBSD ? Shall we add a todo note to write separate tests for it ?

**Comment by siteshwar** [comment]
LGTM

File: src/cmd/ksh93/tests/wchar.sh (hunk 1 above)
Comment (krader1961) [reply src/cmd/ksh93/tests/wchar.sh]: They are supported but for some reason the U+20AC code point can't be translated. Most likely the FreeBSD GB18030 code table is old and missing that character.

Comment (siteshwar) [reply src/cmd/ksh93/tests/wchar.sh]: @saper Do you think it could be a bug in FreeBSD ?

Comment (saper) [reply src/cmd/ksh93/tests/wchar.sh]: Certainly, will have a look!

Comment (krader1961) [reply src/cmd/ksh93/tests/wchar.sh]: @saper You might also want to take a look into why the `LC_ALL` env var doesn't supercede the other locale env vars. Create a file with a UTF-8 encoded U+20AC (the Euro currency symbol). Then run `env LC_ALL=C LC_CTYPE=en_US.UTF-8 wc -m ~/x`. It should report `3` and does on all the other platforms (macOS, OpenBSD, GNU Linux). On FreeBSD 11.1 it reports `1`.

**Pull request closed without merging**